fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/AnalyzeSubquery.java

```diff
@@ -0,0 +1,172 @@
+                            }
+                            return new LogicalSort<>(sort.getOrderKeys(),
+                                    analyzedSubquery(subqueryExprs, sort.child(), ctx.plannerContext));
+                        })
+                )
+        );
+    }
+
+    private List<SubqueryExpr> extractSubquery(Expression expression) {
+        if (expression instanceof SubqueryExpr) {
+            return ImmutableList.of((SubqueryExpr) expression);
+        }
+        Builder<SubqueryExpr> builder = ImmutableList.<SubqueryExpr>builder();
+        getAllSubquery(expression, builder);
+        return builder.build();
+    }
+
+    private void getAllSubquery(Expression expression, Builder builder) {
+        for (Expression expr : expression.children()) {
+            if (expr instanceof SubqueryExpr) {
+                builder.add(expr);
+            } else {
+                getAllSubquery(expr, builder);
+            }
+        }
+    }
+
+    private LogicalPlan analyzedSubquery(List<SubqueryExpr> subqueryExprs, LogicalPlan childPlan, PlannerContext ctx) {
+        for (SubqueryExpr subqueryExpr : subqueryExprs) {
+            if (!subqueryExpr.isAnalyzed()) {
+                if (subqueryExpr instanceof InSubquery) {
+                    return addInSubqueryApplyNodes((InSubquery) subqueryExpr, childPlan, ctx);
+                } else if (subqueryExpr instanceof ScalarSubquery) {
+                    return addScalarSubqueryCorrelatedJoins((ScalarSubquery) subqueryExpr, childPlan, ctx);
+                } else if (subqueryExpr instanceof Exists) {
+                    return addExistsApplyNodes((Exists) subqueryExpr, childPlan, ctx);
+                }
+            }
+        }
+        return childPlan;
+    }
+
+    private LogicalPlan addScalarSubqueryCorrelatedJoins(ScalarSubquery scalarSubquery,
+            LogicalPlan childPlan, PlannerContext ctx) {
+        LogicalPlan enforce = new LogicalEnforceSingleRow<>(scalarSubquery.getQueryPlan());
+        scalarSubquery.setAnalyzed(true);
+        return new LogicalCorrelatedJoin(childPlan, enforce,
+                ctx.getSubquery(scalarSubquery.getQueryPlan()).getCorrelateSlots(),
+                JoinType.INNER_JOIN, Optional.of(BooleanLiteral.TRUE));
+    }
+
+    private LogicalPlan addInSubqueryApplyNodes(InSubquery inSubquery,
+            LogicalPlan childPlan, PlannerContext ctx) {
+        inSubquery.setAnalyzed(true);
+        return appendApplyNode(inSubquery, childPlan,
+                ctx.getSubquery(inSubquery.getQueryPlan()).getCorrelateSlots());
+    }
+
+    private LogicalPlan addExistsApplyNodes(Exists exists,
+            LogicalPlan childPlan, PlannerContext ctx) {
+        exists.setAnalyzed(true);
+        return appendApplyNode(exists, childPlan,
+                ctx.getSubquery(exists.getQueryPlan()).getCorrelateSlots());
+    }
+
+    private LogicalPlan appendApplyNode(SubqueryExpr subqueryExpr, LogicalPlan childPlan,
```

Comment:
this function not necessary, because the body is only one line and method parameter num is same, and this function is not about 'append'